Cargo.toml

```diff
@@ -51,7 +51,6 @@ uint = "0.9.5"
 tempfile = "3.8.1"
 
 [build-dependencies]
```

Comment:
At least I observed issues where `clap_lex` (and likely other clap packages) specified a minimum Rust version (the [is already on 1.74](https://github.com/clap-rs/clap/blob/0cd10d19f0ac39341f89130d924e87879b58867f/Cargo.toml#L17)).

You can see that [clap 4.4.18 fixes it at 1.70](https://github.com/clap-rs/clap/blob/0134f45ff0e2e2be8c451565e4fbf5d3cb7b7cfd/Cargo.toml#L17), whereas [4.5.0 already requires 1.74](https://github.com/clap-rs/clap/blob/a751c5fe65cd33cb09e85ff3039b4fd0182cdb6e/Cargo.toml#L17).

We could consider something less strict like `<4.5.0` instead of `=...` though?

Reply:
I agree that the more liberal we can be with the versioning the better. I don't know what the SOP is for moving the toolchain version forward on the node. That is, when should we be moving the versioning forward vs leaving it the same.  

Seeing as we currently don't use any of these cutting edge `clap` features, I don't see any reason to increment the toolchain version as we don't have a need to do so on the client.

Reply:
As it turns out the best I can do on this is pinning `clap` to `~4.4` and putting `clap_complete` at `<4.5.0`.

Setting `clap` at `<4.5`, for some esoteric reason, results in issues with `clap_lex` specifying Rust version 1.74